_posts/2019-09-20-lambda-to-openfaas.md

```diff
@@ -0,0 +1,330 @@
+---
+title: "Migrate Your AWS Lambda Functions to OpenFaaS"
+description: Burton explains the steps for migrating an existing AWS Lambda function to OpenFaaS
+date: 2019-09-10
+image: /images/lambda-to-openfaas/birds.jpg
+categories:
+  - Lambda
+  - tutorial
+  - examples
+author_staff_member: burton
+dark_background: false
+---
+
+AWS introduced the world to the idea of serverless architecture back in 2014 with the announcement of Lambda. The idea was that you could upload a zip file containing your code logic, and AWS would manage the infrastructure behind the scenes, billing you only for the time the code was being executed.
+
+In his talk [Welcome to Serverless 2.0](https://www.youtube.com/watch?v=JvXm-oHi5Mg), Alex Ellis described the new standard for serverless functions: Docker/OCI images exposing a port for execution, deployed to a Kubernetes cluster. This makes your functions completely portable and removes many of the limitations of the previous generation of serverless functions.
+
+In this post, We'll walk through the steps taken to migrate existing AWS Lambda functions to OpenFaaS compatible functions in order to allow easier local testing, familiar development process, and the freedom to use any number of other cloud providers without any code changes. We won't be moving all of the infrastructure, only the functions. We'll keep the DynamoDB table as well as the IAM roles/permissions to access that data.
```

Comment:
First "We'll" should be "we'll"

Reply:
Say it positive, not negative:

"We won't be moving all of the infrastructure, only the functions."

=> 

"I'll show you how to migrate the compute to OpenFaaS, whilst continuing to use AWS services such as DynamoDB."